File: app/dbmodels/NetworkPort.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'NetworkPort',
    'menu'          => ''
);
$table['fields'] = array(
    'id'                 => DBModels::type('increments',
                                           array('visible' => false)),
    'name'               => DBModels::type('string'),
    'comment'            => DBModels::type('text'),
    'is_deleted'         => DBModels::type('boolean',
                                           array('visible' => false)),
    'is_dynamic'         => DBModels::type('boolean',
                                           array('visible' => false)),
    'asset_id'           => DBModels::type('integer'),
    'networkport_id'     => DBModels::type('integer'),
    'logical_number'     => DBModels::type('integer'),
    'instantiation_type' => DBModels::type('string'),
    'mac'                => DBModels::type('string'),

);

$table['relationships'] = array(
    'asset'     => array(
        'type'  => 'belongsTo',
        'item'  => 'Asset'),
    'network_port' => array(
        'type'  => 'belongsTo',
        'item'  => 'NetworkPort'),
    'network_names' => array(
        'type' => 'morphMany',
        'item' => 'NetworkName'),        
    'vlans'     => array(
        'type'  => 'belongsToMany',
        'item'  => 'Vlan',
        'linktable' => 'network_port_vlans',
        'linkfields' => array('tagged')),
    'network_port_wifi'   => array(
        'type'  => 'hasOne',
        'item'  => 'NetworkPortWifi'),        
    'network_port_local'   => array(
        'type'  => 'hasOne',
        'item'  => 'NetworkPortLocal'),        
    'network_port_ethernet'   => array(
        'type'  => 'hasOne',
        'item'  => 'NetworkPortEthernet'),        
    'network_port_dialup'   => array(
        'type'  => 'hasOne',
        'item'  => 'NetworkPortDialup'),        
    'network_port_aggregate'   => array(
        'type'  => 'hasOne',
        'item'  => 'NetworkPortDialup'),             
    'network_port_alias'   => array(
        'type'      => 'belongsToMany',
        'item'      => 'NetworkPort',
        'linktable' => 'network_port_aliases',
        'field2'    => 'network_port_alias_id'),        
);

include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/networkport.py
from fusionglpi import db


class Networkport(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    is_dynamic = db.Column(db.Boolean, nullable=False, server_default="0")
    asset_id = db.Column(db.Integer, db.ForeignKey('asset.id'))
    networkport_id = db.Column(db.Integer, db.ForeignKey('networkport.id'))
    logical_number = db.Column(db.Integer, nullable=False, server_default="0")
    instantiation_type = db.Column(db.String(255, None, True))
    mac = db.Column(db.String(255, None, True))

    # networkports = db.relationship('Networkport', backref='networkport')
    networkport_aggregates = db.relationship('NetworkportAggregate',
                                             backref='networkport')
    networkport_alias = db.relationship('NetworkportAlias',
                                        backref='networkport')
    networkport_dialups = db.relationship('NetworkportDialup',
                                          backref='networkport')
    networkport_ethernets = db.relationship('NetworkportEthernet',
                                            backref='networkport')
    networkport_locals = db.relationship('NetworkportLocal',
                                         backref='networkport')
    networkport_vlans = db.relationship('NetworkportVlan',
                                        backref='networkport')
    networkport_wifis = db.relationship('NetworkportWifi',
                                        backref='networkport')
